extension_jwt.go

```diff
@@ -0,0 +1,155 @@
+package helix
+
+import (
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+)
+
+// RoleType The user role type
+type roleType string
+
+// Types of user roles used within the JWT Claims
+// TODO expose these when helix supports them
+const (
+	BroadcasterRole roleType = "broadcaster"
+	ExternalRole    roleType = "external"
+	ModeratorRole   roleType = "moderator"
+	ViewerRole      roleType = "viewer"
+
+	toAllChannels = "all"
+)
+
+// PubSubPermissions publish permissions used within
+// JWT claims
+type PubSubPermissions struct {
+	Send   []ExtensionPubSubPublishType `json:"send,omitempty"`
+	Listen []ExtensionPubSubPublishType `json:"listen,omitempty"`
+}
+
+// TwitchJWTClaims contains information
+// containing twitch specific JWT information.
+type TwitchJWTClaims struct {
+	OpaqueUserID string             `json:"opaque_user_id,omitempty"`
+	UserID       string             `json:"user_id"`
+	ChannelID    string             `json:"channel_id,omitempty"`
+	Role         roleType           `json:"role"`
+	Unlinked     bool               `json:"is_unlinked,omitempty"`
+	Permissions  *PubSubPermissions `json:"pubsub_perms"`
+	jwt.StandardClaims
+}
+
+// CreateClaims will construct a claims suitable for generating a JWT token,
+// containing necessary information required by the Twitch API.
+// @param BroadcasterID if this value is empty it will default to 'all'
+// @param pubsub the pubsub permission to attach to the claim
+// @param expiration the epoch of jwt expiration, default 3 minutes from time.Now
+func (c *Client) ExtensionCreateClaims(
+	broadcasterID string,
+	pubsub *PubSubPermissions,
+	expiration int64,
```

Comment:
Presenting and documenting parameters like this is not very idiomatic of Go. I'd prefer we use a struct like this:

```go
type ExtensionCreateClaimsParams struct {
	// BroadcasterID if this value is empty it will default to 'all'
	BroadcasterID string
	// PubSub is the pubsub permission to attach to the claim
	PubSub string
	// Wxpiration is the epoch of jwt expiration, default 3 minutes from time.Now
	Expiration int64
}
```

```go
func (c *Client) ExtensionCreateClaims(params *ExtensionCreateClaimsParams) { ... }
```

Take a read of this article: https://dave.cheney.net/2020/02/23/the-zen-of-go